halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/providers/dcos/DCOSAccountValidator.java

```diff
@@ -0,0 +1,156 @@
+    while (!(parent instanceof DeploymentConfiguration)) {
+      // Note this will crash in the above check if the halconfig representation is corrupted
+      // (that's ok, because it indicates a more serious error than we want to validate).
+      parent = parent.getParent();
+    }
+    deploymentConfiguration = (DeploymentConfiguration) parent;
+    validateClusters(problems, account);
+    if (account.getClusters().isEmpty()) {
+      problems.addProblem(ERROR, "Account does not have any clusters configured")
+          .setRemediation("Edit the account with either --update-user-credential or --update-service-credential");
+    }
+
+    validateDockerRegistries(problems, account, deploymentConfiguration);
+  }
+
+  private void validateClusters(final ConfigProblemSetBuilder problems, final DCOSAccount account) {
+    final NodeIterator children = account.getParent().getChildren();
+
+    Node n = children.getNext();
+    Set<String> definedClusters = new HashSet<>();
+    while (n != null) {
+      if (n instanceof Cluster) {
+        definedClusters.add(((Cluster) n).getName());
+      }
+      n = children.getNext();
+    }
+
+    final Set<String> accountClusters = account.getClusters().stream().map(c -> c.getName())
+        .collect(Collectors.toSet());
+    accountClusters.removeAll(definedClusters);
+    accountClusters.forEach(c -> problems.addProblem(ERROR, "Cluster \"" + c.toString() + "\" not defined for provider")
+        .setRemediation("Add cluster to the provider or remove from the account")
+        .setOptions(Lists.newArrayList(definedClusters)));
+
+    Set<List<String>> credentials = new HashSet<>();
+    account.getClusters().forEach(c -> {
+      final List<String> key = Lists.newArrayList(c.getName(), c.getUid());
+      if (credentials.contains(key)) {
+        problems.addProblem(ERROR,
+            "Account contains duplicate credentials for cluster \"" + c.getName() + "\" and user id \"" + c.getUid()
+                + "\".").setRemediation("Remove the duplicate credentials");
+      } else {
+        credentials.add(key);
+      }
+
+      // TODO(willgorman) once we have the clouddriver-dcos module pulled in we can just validate whether or not
+      // we can connect without a password
+      if (Strings.isStringEmpty(c.getPassword()) && Strings.isStringEmpty(c.getServiceKey())) {
+        problems.addProblem(WARNING,
+            "Account has no password or service key.  Unless the cluster has security disabled this may be an error")
+            .setRemediation("Add a password or service key.");
+      }
+
+      if (!Strings.isStringEmpty(c.getPassword()) && !Strings.isStringEmpty(c.getServiceKey())) {
+        problems.addProblem(ERROR, "Account has both a password and service key")
+            .setRemediation("Remove either the password or service key.");
+      }
+    });
+
+  }
+
+  private void validateDockerRegistries(ConfigProblemSetBuilder psBuilder, DCOSAccount account,
```

Comment:
Could you pull this into a static method somewhere between k8s & dcos? I think this function looks identical (minus some of the DC/OS error message prefixes).